### PR TITLE
Add Apache Cassandra

### PR DESCRIPTION
**Pull request by n0vabyte**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,73 @@
-# apache-cassandra-occ
-Akamai Connected Cloud Apache Cassandra Cluster
+# Linode Marketplace Apache Cassandra Cluster
+
+![cassandra-diagram](images/cassandra_arch.png)
+
+Create a highly available Apache Cassandra cluster using through the Linode Marketplace. Apache Cassandra is an open-source, distributed NoSQL database management system designed for handling large amounts of data across many commodity servers, providing high availability with no single point of failure. Cassandra offers robust support for clusters with asynchronous masterless replication allowing low-latency operations for all clients.
```
Comment (hmorris3293) [inline README.md]:
```suggestion
Create a highly available Apache Cassandra cluster through the Linode Marketplace. Apache Cassandra is an open-source, distributed NoSQL database management system designed for handling large amounts of data across many commodity servers, providing high availability with no single point of failure. Cassandra offers robust support for clusters with asynchronous masterless replication allowing low-latency operations for all clients. 
```

File: provision.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    linode.cloud.instance:
+      label: '{{ instance_prefix }}{{ item | int + 1 }}-{{ uuid }}'
+      api_token: '{{ api_token }}'
+      type: '{{ type }}'
+      region: '{{ region }}'
+      image: '{{ image }}'
+      root_pass: '{{ root_pass }}'
+      authorized_keys: '{{ ssh_keys }}'
+      private_ip: true
+      #stackscript_id: 0
```
Comment (hmorris3293) [inline provision.yml]:
```suggestion
      stackscript_id: 1350852
```

File: roles/apache-cassandra/tasks/ssl.yml [hunk 3]
```diff
@@ -0,0 +1,247 @@
+---
+# create cluster ssl certificates
+
```
Comment (hmorris3293) [inline roles/apache-cassandra/tasks/ssl.yml]:
```suggestion
```

File: roles/apache-cassandra/tasks/ssl.yml [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+    group: cassandra
+    mode: 0640
+    return_content: true
+  register: client_certs
+  with_sequence: count="{{ client_count }}"
+  run_once: true
+  delegate_to: localhost
+
+# generate pkcs12 file from cert keypair
+
```
Comment (hmorris3293) [inline roles/apache-cassandra/tasks/ssl.yml]:
```suggestion
```

File: roles/apache-cassandra/tasks/ssl.yml [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+    common_name: 'client{{ item }}'
+    use_common_name_for_san: false
+    owner: cassandra
+    group: cassandra    
+  with_sequence: count="{{ client_count }}"
+  run_once: true
+  delegate_to: localhost
+
+# sign cert with ca and get client cert
+
```
Comment (hmorris3293) [inline roles/apache-cassandra/tasks/ssl.yml]:
```suggestion
```

File: roles/apache-cassandra/tasks/ssl.yml [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+    path: '{{ cassandra_ssl_key_directory }}/client{{ item }}.key'
+    owner: cassandra
+    group: cassandra
+    mode: 0600
+  with_sequence: count="{{ client_count }}"
+  run_once: true
+  delegate_to: localhost
+
+# generate csrs
+
```
Comment (hmorris3293) [inline roles/apache-cassandra/tasks/ssl.yml]:
```suggestion
```

File: roles/apache-cassandra/tasks/ssl.yml [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+    owner: cassandra
+    group: cassandra
+    mode: 0640
+    return_content: true
+  register: ca_crt
+  run_once: true
+  delegate_to: localhost
+
+# generate ssl keys
+
```
Comment (hmorris3293) [inline roles/apache-cassandra/tasks/ssl.yml]:
```suggestion
```

File: scripts/run.sh [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+  common_name: ${COMMON_NAME}
+  cluster_name: ${CLUSTER_NAME}
+  client_count: ${CLIENT_COUNT}
+  # paths
+  cassandra_config: '/etc/cassandra/cassandra.yaml'
+  cassandra_ssl_path: '/etc/cassandra/ssl'
+  cassandra_cacert: '/etc/cassandra/ssl/ca.crt'
+  cassandra_cakey: '/etc/cassandra/ssl/ca.key'
+  cassandra_cacsr: '/etc/cassandra/ssl/ca-csr'
+  cassandra_truststore: '/etc/cassandra/ssl/cassandra-truststore.jks'
```
Comment (hmorris3293) [inline scripts/run.sh]: do we need these since we use defaults?

Comment (n0vabyte) [reply scripts/run.sh]: We don't need. This should have been removed. Wanna make the suggestion so I can accept into the commit :)

**Comment by hmorris3293** [comment]
redeployed and everything worked as expected 👍

**Pull request merged**